libpod/container_internal.go

```diff
@@ -1515,6 +1515,10 @@ func (c *Container) mountStorage() (_ string, deferredErr error) {
 		}
 
 		mountPoint = overlayMount.Source
+		// change ownership of `merged` created from overlay on top of external rootfs
+		if err := os.Chown(mountPoint, c.RootUID(), c.RootGID()); err != nil {
```

Comment:
I think we need to map the `/merge` directory for the overlay mount (and make sure it is accessible with `makeAccessible`) not the mount point.